pkg/api/v1beta1/conversion.go

```diff
@@ -1054,9 +1063,149 @@ func init() {
 			if err := s.Convert(&in.Secret, &out.Secret, 0); err != nil {
 				return err
 			}
+			if err := s.Convert(&in.AWSElasticBlockStore, &out.AWSElasticBlockStore, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.NFSMount, &out.NFSMount, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.PersistentVolumeClaimVolumeSource, &out.PersistentVolumeClaimVolumeSource, 0); err != nil {
+				return err
+			}
+			return nil
+		},
+		func(in *PersistentVolume, out *newer.PersistentVolume, s conversion.Scope) error {
+			if err := s.Convert(&in.TypeMeta, &out.TypeMeta, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.TypeMeta, &out.ObjectMeta, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Spec, &out.Spec, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Status, &out.Status, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.ClaimRef, &out.ClaimRef, 0); err != nil {
+				return err
+			}
+			return nil
+		},
+		func(in *newer.PersistentVolume, out *PersistentVolume, s conversion.Scope) error {
+			if err := s.Convert(&in.TypeMeta, &out.TypeMeta, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.ObjectMeta, &out.TypeMeta, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Spec, &out.Spec, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Status, &out.Status, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.ClaimRef, &out.ClaimRef, 0); err != nil {
+				return err
+			}
+			return nil
+		},
+		func(in *newer.PersistentVolumeList, out *PersistentVolumeList, s conversion.Scope) error {
```

Comment:
Same for almost all of these.  Conversion functions should be the exception, not the rule - especially for new structs.  The default field-name matching logic should just work.